src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php

```diff
@@ -121,45 +122,14 @@ public function build()
                 $this->webspaces[] = $webspace;
 
                 $this->buildPortals($webspace);
-            } catch (\InvalidArgumentException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidUrlDefinitionException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidCustomUrlException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (WebspaceLocalizationNotUsedException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
+            } catch (\Exception $e) {
+                throw new InvalidWebspaceException(sprintf(
+                    'Could not load webspace "%s"',
+                    $file
+                ), null, $e);
             }
         }
 
-        if (0 === count($this->webspaces)) {
```

Comment:
The idea is that misconfigured webspaces always throw exceptions, so this check would be obsolete in most cases. However, you are right, would still be interesting to see how the system behaves if no webspaces at all are configured...